packages/controllers/src/snaps/endowments/cronjob.test.ts

```diff
@@ -0,0 +1,168 @@
+import { Caveat, PermissionType } from '@metamask/controllers';
+import { SnapCaveatType } from '@metamask/snap-utils';
+import {
+  getCronjobCaveatMapper,
+  cronjobEndowmentBuilder,
+  validateCronjobCaveat,
+  cronjobCaveatSpecifications,
+} from './cronjob';
+import { SnapEndowments } from '.';
+
+describe('endowment:cronjob', () => {
+  describe('specificationBuilder', () => {
+    it('builds the expected permission specification', () => {
+      const specification = cronjobEndowmentBuilder.specificationBuilder({});
+      expect(specification).toStrictEqual({
+        permissionType: PermissionType.Endowment,
+        targetKey: SnapEndowments.Cronjob,
+        endowmentGetter: expect.any(Function),
+        allowedCaveats: [SnapCaveatType.SnapCronjob],
+      });
+
+      expect(specification.endowmentGetter()).toBeUndefined();
+    });
+  });
+
+  describe('cronjobCaveatMapper', () => {
+    it('returns a caveat value for the objects of cronjob specification', () => {
+      expect(
+        getCronjobCaveatMapper({
+          jobs: [
+            {
+              expression: '* * * * *',
+              request: {
+                method: 'exampleMethodOne',
+                params: ['p1'],
+              },
+            },
+            {
+              expression: '* * * * *',
+              request: {
+                method: 'exampleMethodTwo',
+                params: ['p1', 'p2', 'p3'],
+              },
+            },
+          ],
+        }),
+      ).toStrictEqual({
+        caveats: [
+          {
+            type: SnapCaveatType.SnapCronjob,
+            value: {
+              jobs: [
+                {
+                  expression: '* * * * *',
+                  request: {
+                    method: 'exampleMethodOne',
+                    params: ['p1'],
+                  },
+                },
+                {
+                  expression: '* * * * *',
+                  request: {
+                    method: 'exampleMethodTwo',
+                    params: ['p1', 'p2', 'p3'],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      });
+    });
+  });
+});
+
+describe('validateCronjobCaveat', () => {
```

Comment:
Can you test that this works with the object format too?

Reply:
Yes, added object format in the jobs list.